src/components/inspectors/TimerExpression.vue

```diff
@@ -0,0 +1,381 @@
+          <datepicker v-model="endDate" calendar-class="calendar calendaron" :disabled="ends!=='ondate'" format="yyyy-MM-dd"
+                      input-class="form-control end-date"
+                      class=" control float-right"
+                      :class="{'date-disabled' : ends!=='ondate'}"
+                      @selected="updateEndDate"></datepicker>
+        </div>
+        <div class="form-check check-input">
+          <label class="form-check-label">
+            <input type="radio" class="form-check-input" name="optradio" value="after" v-model="ends" @change="update">After &nbsp;
+          </label>
+          <input v-model="times" type="number" min="0" :disabled="ends!=='after'" 
+                 class="form-control control after float-right" @change="update">
+          <label class="occurrences">occurrences</label>
+        </div>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script>
+import Datepicker from 'vuejs-datepicker';
+import moment from 'moment';
+
+const periods = {
+  'day': 'D',
+  'week': 'W',
+  'month': 'M',
+  'year': 'Y',
+};
+
+export default {
+  components: {
+    Datepicker,
+  },
+  props: {
+    value: String,
+    hasEnds: {
+      type: Boolean,
+      default: true,
+    },
+    hasRepeat: {
+      type: Boolean,
+      default: true,
+    },
+    repeatLabel: {
+      type: String,
+      default: 'Repeat every',
+    },
+    weekLabel: {
+      type: String,
+      default: 'Repeat on',
+    },
+  },
+  data() {
+    const date = moment().set('hour', 0).set('minutes', 0);
+    const hours = [];
+    for (let i = 0; i < 48; i++) {
+      hours.push(date.format('HH:mm'));
+      date.add(30, 'minutes');
+    }
+    return {
+      today: moment().format('YYYY-MM-DD'),
+      hours,
+      weekdays: [
+        //  ISO week date weekday number, from 1 through 7,
+        //  beginning with Monday and ending with Sunday.
+        {
+          day: 7,
+          initial: 'S',
+          selected: false,
+        },
+        {
+          day: 1,
+          initial: 'M',
+          selected: false,
+        },
+        {
+          day: 2,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 3,
+          initial: 'W',
+          selected: false,
+        },
+        {
+          day: 4,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 5,
+          initial: 'F',
+          selected: false,
+        },
+        {
+          day: 6,
+          initial: 'S',
+          selected: false,
+        },
+      ],
+      startDate: new Date(),
+      startTime: '00:00',
+      repeat: '1',
+      periodicity: 'week',
+      ends: 'never',
+      endDate: new Date(),
+      times: '1',
+    };
+  },
+  computed: {
+    expression() {
+      this.startDate;
+      this.startTime;
+      this.repeat;
+      this.periodicity;
+      this.selectedWeekdays;
+      this.ends;
+      this.endDate;
+      this.times;
```

Comment:
It was placed to ensure the computed expression updates if any of those attributes change, because since the **makeTimerConfig** has if statements not all the attributes are used at the same time, and during the first development stage some changes were not refreshing the computed, but looks like now is working well without those lines.